src/providers/mssql/qgsmssqldataitems.cpp

```diff
@@ -511,6 +511,12 @@ QString QgsMssqlLayerItem::createUri()
   uri.setUseEstimatedMetadata( QgsMssqlConnection::useEstimatedMetadata( connItem->name() ) );
   mDisableInvalidGeometryHandling = QgsMssqlConnection::isInvalidGeometryHandlingDisabled( connItem->name() );
   uri.setParam( QStringLiteral( "disableInvalidGeometryHandling" ), mDisableInvalidGeometryHandling ? QStringLiteral( "1" ) : QStringLiteral( "0" ) );
+  if ( QgsMssqlConnection::geometryColumnsOnly( connItem->name() ) )
+  {
+    uri.setParam( QStringLiteral( "extentInGeometryColumns" ), QgsMssqlConnection::extentInGeometryColumns( connItem->name() ) ? QStringLiteral( "1" ) : QStringLiteral( "0" ) );
+  }
+  if ( mLayerProperty.isView )
+    uri.setParam( QStringLiteral( "primaryKeyInGeometryColumns" ), QgsMssqlConnection::primaryKeyInGeometryColumns( connItem->name() ) ? QStringLiteral( "1" ) : QStringLiteral( "0" ) );
```

Comment:
I guess this `if ( mLayerProperty.isView )` should be included in the block with `if ( QgsMssqlConnection::geometryColumnsOnly( connItem->name() ) )` ?

Reply:
no, as I say some days ago, if `geometryColumnOnly()` is true, when fetching tables,  QgsMssqlLayerProperty::isView is always false, so layers can't be considered as view.I guess it is because layers referenced in geometry_columns can't be views.

Reply:
Hmm, interesting... Maybe that SQL query does a wrong assumption, that views cannot be listed in `geometry_columns` table - there's probably nothing that would stop users to add views as well... I will check with the clients if they do that. If they do, we probably need to update the SQL query so that instead of always using `0` for "isView" value, this would be checked from MS SQL...